agda/Container/LeafBinaryTree.agda

```diff
@@ -23,6 +25,15 @@ module Container.LeafBinaryTree where
   ... | (ls , rs , _ ) with splitAt (2 ^ k) rs
   ... | (rs , _ , _) = Node (fromVec k ls) (fromVec k rs)
 
+  -- To be improved, we are working with binary trees after all
+  -- It is sad to transform to vectors to lookup, but it's late
+  -- and I don't want to handle decidability cases with 2^k-1
+  lookup-tree : ∀ {A k} → Tree A k → Fin (2 ^ k) → A
```

Comment:
Would it be easier if we were not using a `Fin (2 ^ k)` but an inductive family indexed by `k : Nat` that represent numbers between 0 and 2^k:

    data Finᵇ : Nat → Set where
      here : ∀ {k} → Finᵇ k
      left : ∀ {k} → Finᵇ k → Finᵇ (suc k)
      right : ∀ {k} → Finᵇ k → Finᵇ (suc k)

Going `left` means `+ 0 * 2 ^ k` while going `right` means `+ 1 * 2 ^ k`.

WARNING: untested thought.